modules/combined/test/tests/3d-mortar-projection-tolerancing/test.i

```diff
@@ -0,0 +1,228 @@
+  []
+  [mirror_x]
+    type = ADDirichletBC
+    variable = disp_x
+    boundary = 'mirror_innerp mirror_outerp  '
+    value = 0
+  []
+  [mirror_y]
+    type = ADDirichletBC
+    variable = disp_y
+    boundary = 'mirror_innerp mirror_outerp  '
+    value = 0
+  []
+[]
+
+[Materials]
+  [pellet_properties]
+    type = ADGenericConstantMaterial
+    prop_names = 'density  thermal_conductivity specific_heat'
+    prop_values = '3.3112e3  34 1.2217e3'
+    block = 'pellet_inner pellet_outer'
+  []
+  [pulse_shape_linear]
+    type = ADGenericFunctionMaterial
+    prop_values = '5e10*max(11455*(t)/7,1e-9)'
+    prop_names = 'radial_source'
+    output_properties = 'radial_source'
+    block = 'pellet_inner pellet_outer'
+    use_displaced_mesh = false
+  []
+  [strain]
+    type = ADComputeSmallStrain
+    displacements = 'disp_x disp_y disp_z'
+    eigenstrain_names = eigenstrain #nameS!
+    block = 'pellet_inner pellet_outer '
+  []
+  [thermal_strain]
+    type = ADComputeThermalExpansionEigenstrain
+    stress_free_temperature = ${stress_free_temperature}
+    thermal_expansion_coeff = ${thermal_expansion_coeff}
+    eigenstrain_name = eigenstrain
+    block = 'pellet_inner pellet_outer '
+  []
+  [elasticity]
+    type = ADComputeIsotropicElasticityTensor
+    youngs_modulus = 3.306e11
+    poissons_ratio = 0.329
+  []
+  [stress]
+    type = ADComputeLinearElasticStress
+    block = 'pellet_inner pellet_outer '
+  []
+[]
+
+[Contact]
+  [pellet]
+    primary = void_pellet_0
+    secondary = void_pellet_1
+    model = frictionless
+    penalty = 1e6
+    formulation = mortar
+    c_normal = 1e6
+    correct_edge_dropping = true
+  []
+[]
+
+[UserObjects]
+  [conduction]
+    type = GapFluxModelConduction
+    temperature = T_K
+    boundary = 'void_pellet_0 void_pellet_1    '
+    gap_conductivity = 0.4
+    use_displaced_mesh = true
+  []
+  [rad_pellet]
+    type = GapFluxModelRadiation
+    temperature = T_K
+    boundary = void_pellet_0
+    primary_emissivity = 0.37
+    secondary_emissivity = 0.37
+    use_displaced_mesh = true
+  []
+[]
+
+[Constraints]
+  [gap_pellet]
+    type = ModularGapConductanceConstraint
+    variable = lm_pellet
+    secondary_variable = T_K
+    primary_boundary = 'void_pellet_0'
+    primary_subdomain = pellet_primary_subdomain
+    secondary_boundary = 'void_pellet_1'
+    secondary_subdomain = pellet_secondary_subdomain
+    gap_flux_models = 'conduction rad_pellet' #closed_pellet
+    gap_geometry_type = 'CYLINDER'
+    cylinder_axis_point_1 = '0 0 0'
+    cylinder_axis_point_2 = '0 0 1'
+    use_displaced_mesh = true
+    quadrature = SECOND
+  []
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = NEWTON
+  petsc_options_iname = '-pc_type -pc_factor_mat_solver_package -pc_factor_shift_type -pc_factor_shift_amount '
+  petsc_options_value = 'lu      superlu_dist NONZERO 1e-14'
+  automatic_scaling = true
+  line_search = none
+  ignore_variables_for_autoscaling = 'pellet_normal_lm  lm_pellet '
```

Comment:
I understand that ignoring the scaling of the mechanical contact LM can help due to its high nonlinearity and that mechanical contact typically takes place suddenly. How about the thermal LM: Can scaling be applied to it?

Reply:
Just for my edification.

Reply:
For on-diagonal automatic scaling which is the default, scaling the mechanical LM can be problematic as sometimes there is no on-diagonal. We should be fine to auto-scale with the diagonal of the thermal LM since it does have an on-diagonal. I'll try making that change

Reply:
When scaling thermal LMs, do we need to explicitly specify thermal LMs having their scaling factor computed separately like `scaling_group_variables = 'disp_x disp_y disp_z; T_K; LM1; LM2'` ? Is `scaling_group_variables = 'disp_x disp_y disp_z; T_K'` a **safe** way to let MOOSE knows that we want to compute the LMs variables's scaling factor separately?

Reply:
I had to double check to be sure, but yes it appears that scaling will apply safely to variables not included in the `scaling_group_variables` list. So if pass `Executioner/verbose=true` to the input you gave me, I see this for the scaling factors:
```
Automatic scaling factors:
  disp_x: 2.94915e-09
  disp_y: 2.94915e-09
  disp_z: 2.94915e-09
  T_K: 17.3236
  lm_pellet: 6.51664e+07
  pellet_normal_lm: 1
```

Reply:
> For on-diagonal automatic scaling which is the default, scaling the mechanical LM can be problematic as sometimes there is no on-diagonal. We should be fine to auto-scale with the diagonal of the thermal LM since it does have an on-diagonal. I'll try making that change

I believe scaling happens at each time step (at most). I have had cases where, despite disabling the `compute_only_once` option and using off diagonal terms, if mechanical contact took place in the second or third nonlinear iteration, the simulation would not be able to continue. I had to disable automatic scaling on the mechanical contact LM in some of those cases.